README.md

```diff
@@ -24,3 +25,62 @@ If you want to run the `scons` scripts, you will need a file `data_paths.py` tha
 - `SCRATCH_DIR`: where intermediate output from Gibbs sampling will be sent
 - `CUI_DATA_PATH`: where `partis`-processed data from Cui et al. (2016) lives, if running `get_hedgehog` or `imputed_ancestors_comparison`
 
+## Feature-rich simulation
+
+If you plan on simulating somatic hypermutation on a tree via [gctree](https://github.com/matsengrp/gctree/) or simulating germline sequences via [partis](https://github.com/psathyrella/partis), you will need to run
```

Comment:
If they plan on simulating SHM on a tree, then gctree requires installation using anaconda (pip will not work).

Telling users to install stuff with pip and then installing by anaconda later... seems really inconvenient.
Not sure if we want to just start off in the beginning and recommending users to install by anaconda?

@matsen -- we only need anaconda because gctree wants to import `TreeFace` and some visualization tools (see gctree/bin/gctree.py). However `samm` never uses code paths in gctree that require the visualization tools. It seems very silly to force users to install by anaconda because of code paths we never use. Thoughts? We either tell users to install by anaconda or we fix up the imports in gctree.py.

Reply:
IIUC, you are proposing separating out the simulation from the visualization aspects in gctree.py? 

If so, are you positive that they could be separated? I thought the whole rig uses ete3, which is monolithic and has lots of dependencies.

Reply:
This import line in gctree is the problem: https://github.com/matsengrp/gctree/blob/master/bin/gctree.py#L16
```
from ete3 import TreeNode, NodeStyle, TreeStyle, TextFace, add_face_to_node, CircleFace, PieChartFace, faces, SVG_COLORS
```

Instead, if it said 
```
from ete3 import TreeNode
```
then our stuff would work.
The visualization classes are only used in `render` functions in gctree. However we are only interested in `MutationModel`, which never calls any of the visualization classes and doesnt have a render function (the other tree classes in that file do).

I'm fairly certain if the `MutationModel` class was pulled out to a separate file, we'd be golden.

Reply:
@WSDeWitt does this seem like it will work from your perspective? It would be great for gctree usability.

Reply:
Sure, that seems right.

Reply:
made this change

Reply:
Okay, everything in the simulation scripts work now, but this line

https://github.com/matsengrp/samm/blob/55623b661480fc3290f92cc29cf1a872f16578b7/read_data.py#L202

imports `phylip_parse` which still requires importing `gctree`, which will give this `NodeStyle` error. This function is only used in testing if imputing ancestors is good or not.

Are we cool with this since no one really should be doing imputing or parsing phylip output after we've shown it's not helpful? I think the alternative would be to have `gctree`'s `phylip_parse.py` or `mutation_model.py` have the definitions for `CollapsedTree/Forest` in them instead of having the definitions in `gctree.py`, but this might be annoying for `gctree`ers.

Reply:
Yes it's true that no one should be using the imputing functionality, but it seems like we can make very simple changes to gctree.

Can you add these lines at the top of `phylip_parse.py` instead? It should solve your problem:
```
from gctree import CollapsedTree, CollapsedForest, hamming_distance
```
and then change all the `gctree.___` in `phylip_parse.py` to the actual classes/functions we imported.

Thoughts, @WSDeWitt?

Reply:
I actually tried that and it still gave me the same error. I think so long as we're doing any sort of import from a file that has `from ete3 import NodeStyle` we'll get an error.

Though I just realized `NodeStyle` is required by `CollapsedTree`, so my proposal won't work either :/

Reply:
Oh right...... nvm. this change is not easy. i am okay as is... maybe we can just add a note somewhere in the readme that imputation will require a different installation?

Reply:
Is the most recent commit good, or do we want to be more in-depth?